tests/e2e/vmlistener_test.go

```diff
@@ -120,7 +124,7 @@ func (s *VMListenerTestParams) TestKillVM(c *C) {
 
 	// power on vm
 	govc.PowerOnVM(s.dockerHostName)
-	misc.SleepForSec(35)
+	misc.SleepForSec(45)
```

Comment:
I don't quite follow. We are just waiting for some time to check the VM status, right? We can introduce a util function to periodically call govc.GetVMPowerState(). If this function returns expected state, return immediately; otherwise, sleep for a very short time and retry. This function should return after a maximum limit. Sample code will be like this:

for {
    state := govc.GetVMPowerState(hostName)
    if state == "expected_state" {
        return true;
    }
    if timePassed > maximumLimit {
        return false;
    }
    sleep(3)
}